### PR TITLE
Feature/ha cluster methods

### PR DESCRIPTION
**Pull request by arbulu89**
Depending on the crmsh version we need to use crm or ha-cluster-init/join.

This PR adapts the code to use one or the other depending on the current installed crmsh version.

The rest of the states and formulas are not affected, as the whole management is done in the salt module

### CODE REVIEW

File: salt-saphana.changes [hunk 1]
```diff
@@ -1,3 +1,9 @@
+-------------------------------------------------------------------
+Wed Apr 17 08:28:30 UTC 2019 - Xabier Arbulu Insausti <xarbulu@suse.com>
+
+- Add ha-cluster-init and ha-cluster-join usage when crmsh version
+  is older that 3.0.0
```
Comment (diegoakechi) [inline salt-saphana.changes]:
``` Suggestion
  is older than 3.0.0 
```

File: salt/modules/crmshmod.py [hunk 2]
```diff
@@ -192,20 +281,47 @@ def cluster_init(
...
     '''
-    cmd = '{crm_command} cluster init -y -n {name}'.format(
-        crm_command=CRM_COMMAND, name=name)
+    # INFO: 2 different methods are created to make easy to read/understand
+    # and create the corresponing UT
+    if __crmnewversion__:
+        return _crm_init(
+            name, watchdog, interface, unicast, admin_ip, sbd, sbd_dev, quiet)
+
+    LOGGER.warn('The parameters name and watchdog are not considered!')
```
Comment (diegoakechi) [inline salt/modules/crmshmod.py]: @arbulu89 I think that here, for the older versions, we will need to load the watchdog module manually, to abstract this step to the end user. Even that the ha-cluster scripts didn't cover that at this time, the watchdog configuration was needed. What do you think @krig?

Comment (arbulu89) [reply salt/modules/crmshmod.py]: @diegoakechi Yes, actually we do this in habootstrap-formula.

I didn't want to add this extra logic here

Comment (arbulu89) [reply salt/modules/crmshmod.py]: @diegoakechi What do we have to do for that?
The watchdog module is loaded at least in habootstrap-formula, I don't know if we have to say to the cluster which of the watchdogs to use

Comment (diegoakechi) [reply salt/modules/crmshmod.py]: You need to point SBD what watchdog to use. On the `/etc/sysconfig/sbd` this parameter does that:
`SBD_WATCHDOG_DEV=/dev/watchdog`

If the formula is doing both, that is fine. If not, we need to do it. I think there is a SBD option for that.

Comment (arbulu89) [reply salt/modules/crmshmod.py]: @diegoakechi I will have a look! Ther formula is not doing that, only loading the watchdog. I will need to add this extra step here.

Comment (diegoakechi) [reply salt/modules/crmshmod.py]: You need to configure this before creating the SBD device. So, it should be configured before the sbd create command. @gao-yan Can you confirm that?

Comment (krig) [reply salt/modules/crmshmod.py]: I am not sure `SBD_WATCHDOG_DEV` existed in earlier versions of sbd, might need to look into that.

File: salt/modules/crmshmod.py [hunk 3]
```diff
@@ -154,6 +182,66 @@ def wait_for_startup(
...
+    '''
+    ha-cluster-init command execution
+    '''
+    cmd = '{ha_init_command} -y'.format(
+        ha_init_command=HA_INIT_COMMAND)
+    if interface:
+        cmd = '{cmd} -i {interface}'.format(cmd=cmd, interface=interface)
+    if unicast:
+        cmd = '{cmd} -u'.format(cmd=cmd)
+    if admin_ip:
```
Comment (krig) [inline salt/modules/crmshmod.py]: Older versions of `ha-cluster-init` may lack some of these parameters.. In particular, I think `-S` didn't exist before the move to `crm cluster init`.. but `-u` and `-A` might also be newer.

Comment (arbulu89) [reply salt/modules/crmshmod.py]: I have tested them and in sle12sp1 and sle12sp2 they work, at least `ha-cluster-init` doesn't complain about bad options. But yes, this doesn't mean that they do what they are supposed to do

Comment (diegoakechi) [reply salt/modules/crmshmod.py]: Well, some tests will be needed to check if the options do what they should do... =)

**Comment by arbulu89** [comment]
@diegoakechi @krig I have fount that I need to create more code to manipulate corosync configuration. Old ha-cluster-init package is not setting unicast, so we have to do it manually. I will add some code to do that, test and we will review later again.

**Comment by diegoakechi** [comment]
@arbulu89 please, rebase to avoid conflicts with the changes we did for renaming the project.

**Comment by diegoakechi** [comment]
LGTM.

**Pull request merged**